demos/chat_with_pdf.py

```diff
@@ -22,10 +21,24 @@
 
 from shared_components.cached_llm import CachedLLM
 from shared_components.llm_utils import openai_models
-from shared_components.pdf_utils import (process_file, render_file,
-                                         render_first_page)
+from shared_components.pdf_utils import process_file, render_file, render_first_page
 from shared_components.theme_management import load_theme
 
+from langchain.chains import create_retrieval_chain
+from langchain.chains.combine_documents import create_stuff_documents_chain
+
+from datasets import Dataset
+
+from ragas.metrics import (
+    faithfulness,
+    answer_relevancy,
+)
+
+from ragas import evaluate
+
+# this isn't supported yet with new version https://stackoverflow.com/questions/77982768/no-module-named-ragas-langchain
```

Comment:
see issue but RAGAS thing is behind so rolled it myself

Reply:
Nice!